airflow/models.py

```diff
@@ -2010,6 +2129,105 @@ def get(cls, key, session, deserialize_json=False):
         return v
 
 
+class XCom(Base):
+    """
+    Base class for XCom objects.
+    """
+    __tablename__ = "xcom"
+
+    id = Column(Integer, primary_key=True)
+    key = Column(String)
+    value = Column(PickleType(pickler=dill))
+    timestamp = Column(DateTime, server_default=func.current_timestamp())
+    visible_on = Column(DateTime, nullable=False)
+
+    # source information
+    task = Column(String, nullable=False)
+    dag = Column(String, nullable=False)
+
+    def __repr__(self):
+        return '<XCom "{key}" ({task} @ {visible_on})>'.format(
+            key=self.key,
+            task=self.task,
+            visible_on=self.visible_on)
+
+    @classmethod
+    @provide_session
+    def set(
+            cls,
+            key,
+            value,
+            visible_on,
+            task,
+            dag,
+            session=None):
+        """
+        Store an XCom value.
+        """
+
+        if not isinstance(task, basestring) or not isinstance(dag, basestring):
```

Comment:
You can also let the underlying library handle this and fail, but type checking isn't a bad thing to have...